walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -138,4 +140,57 @@
 
         return ApiFutures.transform(snapshotFuture, jobFunction, MoreExecutors.directExecutor());
     }
+
+    /** Returns future of all ACTIVE and eligible job posts in database. */
+    public Future<Collection<Job>> fetchAllEligibleJobs(List<String> skills) throws IOException {
+
+        final Query activeJobsQuery = FireStoreUtils.getFireStore()
+                .collection(JOB_COLLECTION)
+                .whereEqualTo(JOB_STATUS_FIELD, JobStatus.ACTIVE);
+
+        ApiFuture<QuerySnapshot> querySnapshotFuture = activeJobsQuery.get();
+
+        ApiFunction<QuerySnapshot, Collection<Job>> function = documents -> {
+            ImmutableSet.Builder<Job> jobs = ImmutableSet.builder();
+
+            for (DocumentSnapshot document : documents) {
+                Job job = document.toObject(Job.class);
+                List<String> requirements = job.getRequirements();
+
+                // Eligible job: all job requirements are contained in the applicant skills.
+                if (skills.containsAll(requirements)) {
```

Comment:
There should be a way to do this before getting all the documents from the database. Something like `array-contains`

Reply:
i tried to use `array-contains`, but this is `OR` operation, what i need is `AND` operation😂
and also cloud firestore can only make `requirement containsAll skills`, but what i need is `skills containsAll requirement`

Reply:
Should we change array to dictionary?
https://stackoverflow.com/questions/54987399/firestore-search-array-contains-for-multiple-values